docs/contributing.md

```diff
@@ -2,6 +2,12 @@
 
 We welcome contributions from the community! This guide will help you understand our contribution process and requirements.
 
+## Development guidelines
+
+1. Small PRs ([blogpost](https://testing.googleblog.com/2024/07/in-praise-of-small-pull-requests.html))
+1. If you are fixing a bug, write a test reproducing a bug (the test should fail without your changes)
+1. If you are refactoring, make sure there are tests covering this area, if not - introduce tests first, then continue with refactoring
```

Comment:
Also, all those points are `1.`

Reply:
yea kind of same question from me, seems like tests in separated PR sounds better but there are some cons to it: 
- it needs to be merged before fix
- after merge, fix needs to be also created pretty quickly as then develop branch is 'blocked' in case of some other fix releases (not often thing but seems worth considering during process creation)

on the other hand when doing it in one PR: 
- PR are bigger
- harder to compare before and after fix

so I would say one PR is safer but many PRs are better if we ignore urgent releases.

I would personally go with many as i think we can somehow deal with 'emergencies'. 

Plus I would change wording a bit 
`if not - introduce tests first, then start with refactoring`

Reply:
@kkedziak-splunk @soleksy-splunk feel free to propose changes to the content and the wording.

Reply:
updated, please review again